rskj-core/src/main/java/co/rsk/trie/NodeReference.java

```diff
@@ -51,6 +51,7 @@ public boolean isEmpty() {
     /**
      * The node or empty if this is an empty reference.
      * If the node is not present but its hash is known, it will be retrieved from the store.
+     * If the node could not be retrieved from the store, a NodeReferenceException is thrown.
      */
     public Optional<Trie> getNode() {
```

Comment:
is this only used from `MinerUtils`? just to understand if we have to modify the `catch` clauses on any other class apart from that

Reply:
This method is used in more than one place, but the exception _per se_ is only being captured in MinerUtils and given that `NodeReferenceException` extends from `RuntimeException` there shouldn't be any problem in the rest of them.

Reply:
awesome!

Reply:
Strategy was updated. Dismissed.